src/plugins/DevTools/components/ActionsComponent.ts

```diff
@@ -1,36 +1,106 @@
+    selectedRegistryRecordIndex: number;
 }
 
 export class ActionsComponent extends OwlUIComponent {
     state: ActionsState = {
-        selectedIndex: 0, // Index of the selected action in the stack
+        actions: [], // Stack of all actions performed since init
+        openTab: 'selected',
+        registry: this.env.editor.dispatcher.registry,
+        selectedActionIndex: null, // Index of the selected action in the stack
+        selectedHandlerToken: null,
+        selectedRegistryRecordIndex: null, // Index of the selected registry record
     };
 
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Add an action to the stack
+     *
+     * @param {Action} action
+     */
+    addAction(action: Action): void {
+        this.state.actions.unshift(action);
+        this.selectAction(0);
+    }
+    /**
+     * Return the action handler corresponding to the given token.
+     *
+     * @param {HandlerToken} handlerToken
+     * @returns {ActionHandler|null}
+     */
+    getHandler(handlerToken: HandlerToken): ActionHandler | null {
+        const record = this.state.registry.find((record: DispatcherRegistryRecord): boolean => {
+            return !!record.handlers[handlerToken];
+        });
+        return (record && record.handlers[handlerToken]) || null;
+    }
     /**
      * Handle keydown event to navigate in the action stack
      */
     onKeydown(event: KeyboardEvent): void {
         if (event.code === 'ArrowDown') {
-            this.state.selectedIndex += 1;
+            this.state.selectedActionIndex += 1;
         } else if (event.code === 'ArrowUp') {
-            this.state.selectedIndex -= 1;
+            this.state.selectedActionIndex -= 1;
         } else {
             return;
         }
         event.preventDefault();
         event.stopImmediatePropagation();
     }
+    /**
+     * Open the tab with the given `tabName`
+     *
+     * @param {'selected' | 'registry'} tabName
+     */
+    openTab(tabName: 'selected' | 'registry'): void {
+        this.state.openTab = tabName;
+    }
     /**
      * Select the action at given index
      *
      * @param {number} index
      */
     selectAction(index: number): void {
-        this.state.selectedIndex = index;
+        this.state.selectedActionIndex = index;
+        const action: Action = this.state.actions[index];
+        const correspondingRecordID: number | null = action
+            ? this._getRegistryRecordIndex(action)
+            : null;
+        this.selectRegistryRecord(correspondingRecordID);
+    }
+    /**
+     * Select the action handler corresponding to the given token.
+     *
+     * @param {HandlerToken|null} handlerToken
+     */
+    selectHandler(handlerToken: HandlerToken | null): void {
+        this.state.selectedHandlerToken = handlerToken;
```

Comment:
We need it so it can be called in the xml template. Now where I use `this.selectHandler(null)`, nothing keeps me from using `this.state.selectedHandlerToken = null` but personally I'd rather use the method in case later we decide that something else that just `this.state.selectedHandlerToken = null` needs to be done when selecting a handler. I'd rather not scatter the code.

Reply:
If we choose to keep the method then yes, we indeed should continue to use it.